[Prometheus ] Add remote_write documentation

tetianakravchenko opened the pull request:
## What does this PR do?

add documentation regarding the remote_write data_stream:
- create service when running agent in kubernetes
- add information regarding `write_relabel_configs` to align with https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-metricset-prometheus-remote_write.html

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

<!-- Recommended
Add a checklist of things that are required to be reviewed in order to have the PR approved
-->
- [ ]

## How to test this PR locally

<!-- Recommended
Explain here how this PR will be tested by the reviewer: commands, dependencies, steps, etc.
-->

## Related issues

<!-- Recommended
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.

- Closes #123
- Relates #123
- Requires #123
- Supersedes #123
-->
-

## Screenshots

<!-- Optional
Add here screenshots presenting:
- Kibana UI forms presenting configuration options exposed by the integration
- dashboards with collected metrics or logs
-->

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-5356/detail/PR-5356/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-5356/detail/PR-5356/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-5356/detail/PR-5356/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-5356/detail/PR-5356/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_5356.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/fleet-ci/transactions/view?rangeFrom=2023-02-22T13:23:41.783Z&rangeTo=2023-02-22T13:43:41.783Z&transactionName=BUILD+Ingest-manager%2Fintegrations%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=76d75c6ecb089bd7bdcaea989a7ccd8d&transactionId=8547e9428c311b70)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-02-22T13:33:41.783+0000


* Duration: 14 min 15 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 9  |
| Skipped      | 0 |
| Total        | 9   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->














## :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_comment.id-->

elasticmachine commented:
### :globe_with_meridians: Coverage report

Name | Metrics % (`covered/total`) | Diff
 --- | --- | ---
Packages | 100.0% (`0/0`) | :green_heart: 
Files | 100.0% (`0/0`) | :green_heart: 
Classes | 100.0% (`0/0`) | :green_heart: 
Methods | 88.889% (`8/9`) | :-1: -11.111
Lines | 100.0% (`0/0`) | :green_heart: 0.82
Conditionals | 100.0% (`0/0`) | :green_heart: 

<!--COMMENT_GENERATED_WITH_ID_coverage-->

gizas commented:
Not related to this context but because I read again the doc, 

1. Can we make links the following?
```
Prometheus Exporters (Collectors)
Prometheus Server Remote-Write
Prometheus Queries (PromQL)
```
Also above be a numeric list?

2. Change first header : Collector Metrics -> Prometheus Exporters (Collectors)

3. Make all headers below :
1. Prometheus Exporters (Collectors)
2. Prometheus Server Remote-Write
3. Prometheus Queries (PromQL)

tetianakravchenko commented:
@gizas thank you for the review!
I've renamed headers and added links in this [commit](https://github.com/elastic/integrations/pull/5356/commits/9a5264b03771d70bd344af277b258a8933b51454).

> Can we make links the following?

tbh, I am not sure how added links will affect documentation [here](https://docs.elastic.co/integrations/prometheus), but it at least simplify in repository

> Also above be a numeric list?

I don't think it should be a numeric list, as on this [page](https://docs.elastic.co/integrations/prometheus) it might affect content of `On this page` on the right
